tests/test_quoteengine.py

```diff
@@ -0,0 +1,124 @@
+from datetime import datetime
+from unittest.mock import patch
+from unittest.mock import MagicMock
+
+import pytest
+
+from tululbot.utils import QuoteEngine
+
+
+class FakeResponse:
+    def get_json(self):
+        return self.json
+
+    pass
+
+
+@pytest.fixture
+def qe():
+    return QuoteEngine()
+
+
+def cdn_with_branch(branch):
+    return 'https://cdn.rawgit.com/tulul/tulul-quotes/{}/quote.yaml'.format(branch)  # noqa
+
+
+def test_refresh_time(qe):
+    with patch('tululbot.utils.TimeHelper') as mock_dt:
+        with patch('tululbot.utils.requests') as mock_requests:
+            response1 = FakeResponse()
+            response1.status_code = 200
+            response1.json = {
+                "commit": {
+                    "sha": "beefbeef"
+                }
+            }
+
+            response1.ok = True

```

Comment:
I see. so you're assigning the properties here. can you please put them as instance variable and pass them to the constructor instead? that way it's easier to understand what a fake response has.

in general I'm not a fan of dynamically attaching properties like this. often times this leads to confusion like "where the hell this property comes from if it's not defined inside the class definition?"